internal/graph/checkdispatchset_test.go

```diff
@@ -0,0 +1,387 @@
+				tuple.MustParse("document:somedoc#viewer@anothertype:2#member"),
+				tuple.MustParse("document:somedoc#viewer@anothertype:3#member"),
+			},
+			2,
+			[]checkDispatchChunk{
+				{
+					resourceType:       relationRef{namespace: "group", relation: "member"},
+					resourceIds:        []string{"1", "2"},
+					hasIncomingCaveats: false,
+				},
+				{
+					resourceType:       relationRef{namespace: "group", relation: "member"},
+					resourceIds:        []string{"3"},
+					hasIncomingCaveats: false,
+				},
+				{
+					resourceType:       relationRef{namespace: "anothertype", relation: "member"},
+					resourceIds:        []string{"1", "2"},
+					hasIncomingCaveats: false,
+				},
+				{
+					resourceType:       relationRef{namespace: "anothertype", relation: "member"},
+					resourceIds:        []string{"3"},
+					hasIncomingCaveats: false,
+				},
+			},
+			map[string][]resourceIDAndCaveat{
+				"group:1#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+				"group:2#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+				"group:3#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+				"anothertype:1#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+				"anothertype:2#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+				"anothertype:3#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+			},
+		},
+		{
+			"some caveated members",
+			[]*core.RelationTuple{
+				tuple.MustParse("document:somedoc#viewer@group:1#member[somecaveat]"),
+				tuple.MustParse("document:somedoc#viewer@group:2#member"),
+				tuple.MustParse("document:somedoc#viewer@group:3#member"),
+			},
+			100,
+			[]checkDispatchChunk{
+				{
+					resourceType:       relationRef{namespace: "group", relation: "member"},
+					resourceIds:        []string{"1", "2", "3"},
+					hasIncomingCaveats: true,
+				},
+			},
+			map[string][]resourceIDAndCaveat{
+				"group:1#member": {
+					{resourceID: "somedoc", caveat: caveatForTesting("somecaveat")},
+				},
+				"group:2#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+				"group:3#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+			},
+		},
+		{
+			"caveated members combined when chunking",
+			[]*core.RelationTuple{
+				tuple.MustParse("document:somedoc#viewer@group:1#member[somecaveat]"),
+				tuple.MustParse("document:somedoc#viewer@group:2#member"),
+				tuple.MustParse("document:somedoc#viewer@group:3#member"),
+				tuple.MustParse("document:somedoc#viewer@group:4#member[somecaveat]"),
+			},
+			2,
+			[]checkDispatchChunk{
+				{
+					resourceType:       relationRef{namespace: "group", relation: "member"},
+					resourceIds:        []string{"2", "3"},
+					hasIncomingCaveats: false,
+				},
+				{
+					resourceType:       relationRef{namespace: "group", relation: "member"},
+					resourceIds:        []string{"1", "4"},
+					hasIncomingCaveats: true,
+				},
+			},
+			map[string][]resourceIDAndCaveat{
```

Comment:
why isn't `group:4#member` showing up here?

Reply:
Forgot to add it and the test only verifies what I put in that map